allow underscores in query variables

mnazarov opened the pull request:
## Description

* modified regex to find variables to match the spec - https://spec.graphql.org/October2021/#sec-Language.Variables
* added test with variables
* minor typo fix

## Example
<!--- if introducing a new feature or changing behavior of existing
methods/functions, include an example if possible to do in brief form -->
Previously one couldn't use underscores in the variable names, now it is possible:
```
qry <- Query$new()$query('repos', '
  query getRepo($owner_name: String!, $repo: String!) {
    repository(owner: $owner_name, name: $repo) {
      name
    }
  }
')
```
See the test code for full example - https://github.com/mnazarov/ghql/blob/master/tests/testthat/test-GraphqlClient.R#L131-L148

mpadge commented:
Thanks @mnazarov. Before merging, we need to fix the check failures. They're not your fault, but just because this package hasn't been touched for a while (Scott has left rOpenSci, and I've taken over maintenance of this package.) Could you please merge the master branch with your local branch and push those changes here? Tests should then pass, and we can merge.

mnazarov commented:
Thanks @mpadge for picking this up and providing details. I have now merged and fixed tests so that all the checks pass.

Note that I also added another change in the meantime - _which probably would have been better in a separate PR_ - relax check for variables by allowing NULL's.
This is an intermediate solution between current behaviour where it stops whenever a variable is not provided, and completely removing [the check](https://github.com/ropensci/ghql/blob/6352c8e1dc089045f652f98a75e7664e963c82f7/R/client.R#L291-L293). The idea behind is that some variables may be optional in the query, so we should allow that too (with my PR you still need to pass a NULL).
Ideally we'd need to detect the type of the variable and check it only if this type is required (i.e. marked as NonNull with `!`)

mpadge commented on R/client.R:
```diff
@@ -247,7 +247,7 @@ GraphqlClient <- R6::R6Class(
   private = list(
     # @field .var_regex variable regexp 
-    .var_regex = '\\$([[:alnum:]]+)',
+    .var_regex = '\\$([_A-Za-z][_0-9A-Za-z]*)',
```
@mnazarov I'd suggest a few tweaks to the regex:
1. I find regex's more comprehensible with named classes, so would personally prefer `[[:alpha:]]` and `[[:alphanum:]]` instead of `[A-z]` and the like
2. That said, note that in R, [`\w` (word) means `[[:alnum:]_]` anyway](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/regex), so you could just use that.
3. That means you can just reduce the whole lot to `\\$(\w+)`
4. Finally, you've changed the final `+` to `*` which means it could still match an empty string, whereas the former version with `+` seems more appropriate as it presumes a string of non-zero length. I'd suggest to revert to `+`, but happy to hear your opinions on that.

mnazarov replied:
@mpadge thanks for suggestions! Please note that here I used 2-part expression - first only for the 1st symbol (`[_A-Za-z]`) and then all the rest (`[_0-9A-Za-z]*`) - this is needed because the first symbol cannot be a number. That's why `\\$(\w+)` wouldn't work, and why I've put a `*` in the second part - for the case of one-letter variables.
But indeed the regex can be simplified to `\\$([[:alpha:]_]\\w*)` - I just added a commit with that
